functions/venue.js

```diff
@@ -711,31 +711,42 @@ exports.deleteTable = functions.https.onCall(async (data, context) => {
   const tables = docData.config.tables || defaultTables;
 
   const index = tables.findIndex((val) => val.reference === tableName);
+
+  tables.splice(index, 1);
+
+  const recalculatedTables = tables.map((table, i) => ({
+    ...table,
+    reference: `Table ${i + 1}`,
```

Comment:
Not sure why this `reference` is needed and more important where.
Is it breaking some cloud function? Maybe that one should be updated. 
Is it used in other places? Maybe even the simulator.

Ideally tables would have `id` or Firestore key to reference them, but in this case, if only their position in an array is relevant, the change of the `reference` value is either not necessary or used in specific places